// PB definitions for CityLex.

syntax = "proto2";

package citylex;

// A series of values paired with a lemma or wordform.
message Entry {

  // Frequencies.

  optional int64 celex_freq = 1;

  optional int64 subtlex_uk_freq = 2;
  optional int64 subtlex_uk_cd = 3;

  optional int64 subtlex_us_freq = 4;
  optional int64 subtlex_us_cd = 5;

  // Morphological analyses.

  message MorphTriple {
    required string lemma = 1;
    optional string features = 3;
  }

  message Segmentation {
    required string segmentation = 1;
    required int32 nmorph = 2;
  }

  repeated MorphTriple celex_morph = 6;

  optional string elp_morph_sp = 7;
  optional int32 elp_nmorph = 8;

  repeated MorphTriple udlexicons_morph = 9;

  repeated MorphTriple unimorph_morph = 10;

  // Pronunciations.

  repeated string celex_pron = 11;

  repeated string cmudict_pron = 12;

  repeated string wikipron_uk_pron = 13;

  repeated string wikipron_us_pron = 14;
}

// A collection of entries.
message Lexicon {
  map<string, Entry> entry = 1;
}
